bpf/Makefile

```diff
@@ -20,7 +20,7 @@ DEPS = $(patsubst %,$(IDIR)/%,$(_DEPS))
 
 FLAGS := -I. -I/usr/include/x86_64-linux-gnu/ \
 	 -Wall -Werror \
-	 -Wno-address-of-packed-member -Wno-compare-distinct-pointer-types -Wno-unknown-warning-option \
+	 -Wno-address-of-packed-member -Wno-compare-distinct-pointer-types -Wno-unknown-warning-option -Wno-unused-but-set-variable \
```

Comment:
Why has 78a36b3 been reverted by 95417d3? That seems to be the proper fix, not silencing the warning.
CC: @willfindlay

Reply:
@sh7dm I reverted my entire PR because we noticed it broke under some kernels during testing. It's been re-opened but the bug hasn't been fixed yet.

Reply:
Let's not silence the warning, the unused variables should just be removed. Feel free to cherry-pick from my new PR.

Reply:
@sh7dm I haven't created the roadmap list yet, but we are working on getting go tests running on a lot of different kernels from the github actions. We have something internally, but its a bit tricky to get it hooked up here. Coming soon though.

Reply:
Sorry, is it related to this thread about the warning?

Reply:
@sh7dm It's about the test failure I mentioned. This will work on recent kernels but some old ones will still reject the programs. We're working on adding multi-kernel testing to OSS so we can catch this in CI. (We already have something internally but it's a matter of getting it hooked up here.)